### PR TITLE
Changes js error show with logblacklist

### LINKED ISSUE #4711: [Bug] derived property errors are shown in debugger
## Description

Derived property error are shown in debugger. eg for table, The value at selectedRows is invalid error is shown 
### Steps to reproduce the behaviour:

Add steps to reproduce this behaviour, include console / network logs & screenshots

1. Go to any application
2. Add table widget
3. check debugger for 'The value at selectedRows is invalid' error

image

### Important Details

- Version: [Cloud / Self-Hosted vx.x]
- OS: [e.g.MacOSX]
- Environment [production, release, deploy preview]

### LINKED ISSUE #4720: [Bug] Expected Data Type is not displayed to user on an error in the API pane
Steps to reproduce:
1) Add an API in Pane
2) Ensure to add inappropriate data 
3) Click on the body 
and observe the property pane 

Observation:
User is displayed with evaluated data on both the place along with "debug" option

Expectation:
User must be displayed with Expected Data Type along with "debug" option

Screenshot 2021-05-27 at 4 01 49 PM

### LINKED ISSUE #4773: [Bug] Query pane displays an error when Query is added
Steps to reproduction:
1) Navigate to a Query Pane 
2) Click on Datasource
3) Add a Query with smart substitution 
and observe

Observation:
It is observed that an error is displayed to the user and observe that clicking on the Debug button the error is not recorded

Expectation:
User must not be displayed with an error message when an appropriate query is added and execution is  correct

Screenshot 2021-05-28 at 2 00 04 PM

### LINKED ISSUE #4820: [Bug] The error does not get truncated in header parameter
Steps to reproduce:
1) Navigate to API 
2) Add an API 
3) Now add value and Key in the header (long value)
4) Observe the expected value 

Observation:
When the value is long it is observed that it does not get truncated 

Expectation
It is expected to get truncated when the value is long 

Screenshot 2021-06-01 at 9 57 46 AM

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description
Refactor code to shos js erros in debugger

Fixes #4711 
Fixes #4720 
Fixes #4773
Fixes #4820 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: changes-js-error-show-with-logblacklist 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 51.59 **(0.01)** | 32.7 **(0.05)** | 29.72 **(-0.02)** | 52.1 **(0)**
 :green_circle: | app/client/src/components/editorComponents/CodeEditor/index.tsx | 75.23 **(0.96)** | 52.94 **(1.64)** | 58.06 **(0)** | 75.88 **(0.75)**
 :green_circle: | app/client/src/entities/DataTree/dataTreeWidget.ts | 84.85 **(1.52)** | 25 **(0)** | 80 **(5)** | 84.38 **(1.62)**
 :red_circle: | app/client/src/mockResponses/WidgetConfigResponse.tsx | 12.09 **(-1.83)** | 0 **(0)** | 0 **(0)** | 12.5 **(-1.97)**
 :red_circle: | app/client/src/sagas/ActionSagas.ts | 12.01 **(-0.05)** | 1.06 **(0)** | 4.55 **(0)** | 14.59 **(-0.07)**
 :green_circle: | app/client/src/sagas/DebuggerSagas.ts | 18.06 **(4.01)** | 4.35 **(1.36)** | 33.33 **(8.33)** | 19.12 **(3.52)**
 :red_circle: | app/client/src/sagas/EvaluationsSaga.ts | 19.19 **(-1.5)** | 1.56 **(0)** | 9.09 **(-0.91)** | 22.15 **(-1.29)**
 :red_circle: | app/client/src/utils/DynamicBindingUtils.ts | 88.24 **(-0.22)** | 75.41 **(0)** | 78.57 **(0)** | 86.81 **(-0.29)**
 :red_circle: | app/client/src/utils/WidgetPropsUtils.tsx | 63.98 **(-1.62)** | 52.32 **(-1.06)** | 54.17 **(-3.18)** | 63.34 **(-1.68)**
 :green_circle: | app/client/src/workers/DataTreeEvaluator.ts | 88.71 **(0.34)** | 78.38 **(1.42)** | 87.32 **(0)** | 88.79 **(0.37)**</details>

### CODE REVIEW

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 1]
```diff
@@ -41,14 +41,136 @@ import {
...
+    const results = yield call(
+      getNewErrors,
+      dataTree,
+      evaluationOrder,
+      debuggerErrors,
+    );
+    finalErrors = [...finalErrors, ...results];
+  }
+
+  if (!finalErrors) return;
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:172]: Dont return here. We still want to process other errors

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 2]
```diff
@@ -41,14 +41,136 @@ import {
...
+  finalErrors.forEach((error) => {
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:173]: Loop through original errors and not final errors

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 3]
```diff
@@ -104,6 +226,7 @@ const evalErrorHandler = (errors: EvalError[]) => {
         });
         break;
       }
+      // step 3 : remove these
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:229]: Forgot this step

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 4]
```diff
@@ -156,10 +279,19 @@ function* evaluateTreeSaga(
       widgetTypeConfigMap,
     },
   );
-  const { dataTree, dependencies, errors, logs } = workerResponse;
+  // step 7: get evaluationOrder
```
Comment (hetunandu) [inline app/client/src/sagas/EvaluationsSaga.ts:282]: Remove the step comments

File: app/client/src/workers/DataTreeEvaluator.ts [hunk 5]
```diff
@@ -595,6 +603,14 @@ export default class DataTreeEvaluator {
     }
     return undefined;
   }
+  //check if the property is part if logBlackList object to show
+  //error in debugger
+  isLoggingBlocked(entity: DataTreeWidget, propertyPath: string) {
+    if (entity.logBlackList) {
+      return !!entity.logBlackList[propertyPath];
+    }
+    return false;
```
Comment (hetunandu) [inline app/client/src/workers/DataTreeEvaluator.ts:612]: Unused function

**Review by hetunandu — CHANGES_REQUESTED** [review]

File: app/client/src/sagas/EvaluationsSaga.ts (hunk 4 above)
Comment (ApekshaBhosale) [reply app/client/src/sagas/EvaluationsSaga.ts:282]: i will remove once we are done with all the changes

**Review by hetunandu — DISMISSED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**